Refresh: update Edge doc with Digital Ocean managed K8s example (#54)
crate/cloud-getting-started: docs/edge.rst

.. _edge-tools:

> Install CrateDB Edge using third-party tools

Reviewer: ```suggestion
Install CrateDB Edge using managed k8s
```

============================================

> You can combine CrateDB Edge with third-party tools for ease of use, such as

Reviewer: ```suggestion
You can combine CrateDB Edge with tools for ease of use, such as
```

> managed Kubernetes providers and preconfigured Kubernetes distributions. In the

Reviewer: ```suggestion
managed Kubernetes providers and self-hosted Kubernetes setups. In the
```

former category, we have tested AWS, Azure, Digital Ocean, and GCP with the
CrateDB Edge stack, and for the latter the distributions described below. In
the walkthroughs below, we provide by way of example a guide for using
`Digital Ocean`_ as a managed Kubernetes provider with CrateDB Edge, and guides

> for two of the most common Kubernetes distributions: `Microk8s`_ and `K3s`_.

Reviewer: ```suggestion
for two of the most common Kubernetes distributions: `microk8s`_ and `k3s`_.
```

Author: They write it as "MicroK8s" and "K3s" themselves so I would follow that actually

    These guides are provided as example scenarios only. Other managed
    Kubernetes providers or preconfigured Kubernetes distributions may also
    work with CrateDB Edge.

These are third-party tools and Crate.io is not responsible for them. That
said, we have tested the instructions provided below for functionality. Users
less familiar with customizing their Kubernetes stack on their own may find
either of these approaches a practical solution for easier CrateDB Edge setup.

.. _edge-tools-digitalocean:

Digital Ocean
-------------

Below is a step-by-step guide to using Digital Ocean as a managed Kubernetes
provider in combination with CrateDB Edge. The steps are merely examples of a
process validated by us; other methods may work also. We provide this
information for ease of use and to illustrate how to work with CrateDB Edge.

Signup
''''''

First you must sign up with `Digital Ocean`_. On the Kubernetes page, click
*Sign up* and make an account. Verify your email address to proceed. (Digital
Ocean may also require a token pre-payment.)

Create Kubernetes cluster
'''''''''''''''''''''''''

Create a Kubernetes cluster using the Digital Ocean cloud interface, under

> "Manage", then "Kubernetes". When configuring the machine, make sure to choose

Reviewer: ```suggestion
"Manage", then "Kubernetes". When configuring the cluster, make sure to choose
```

download the configuration YAML file and note the local address of the file.
Install `kubectl`_ if you have not done so already. Then point the Kubeconfig
configuration path to where you stored the YAML file:

.. code-block:: console

    export KUBECONFIG=~<file source>

Subsequently, wait for the install to finish and check that the nodes are
running as intended:

.. code-block:: console

    kubectl get nodes

Set up Edge region
''''''''''''''''''

Now, go to the CrateDB Cloud Console and create a custom CrateDB Edge region.
Follow the steps outlined :ref:`from the CrateDB sign up onwards <edge-signup>`
to proceed. Run the script the CrateDB Cloud Console shows in the panel for the
custom region you just created and install prerequisites as necessary.

Define storage class
''''''''''''''''''''

Eventually, the script will indicate that there is no ``crate-premium`` storage
class available. To define this storage class correctly, copy the default
storage class Digital Ocean provides, then change the the ``name`` to
``crate-premium`` in the copied file. For example, using kubectl and Vim:

.. code-block:: console

    kubectl get sc do-block-storage -o yaml > sc.yaml

>     vi sc.yaml

Reviewer: ```suggestion
  kubectl get sc do-block-storage -o yaml | grep -vi is-default-class | sed -e 's/name: do-block-storage/name: crate-premium/' | kubectl create -f -
```
